Implement stability test (#3136)
GEOS-DEV/GEOS: src/coreComponents/constitutive/docs/CompositionalMultiphaseFluid.rst

    y_{1} & y_{2} & y_{3} & \cdots & y_{N_c} \\
    \end{bmatrix}

where :math:`x_c` is the mole fraction of component :math:`c` in the liquid phase and :math:`y_c`
is the mole fraction of component :math:`c` in the vapor phase.

The fluid properties are updated through the following steps:

1) The phase fractions (:math:`\nu_p`) and phase component fractions (:math:`x_c` and :math:`y_c`) are
computed as a function of pressure (:math:`p`), temperature (:math:`T`) and total component fractions
(:math:`z_c`).

2) The phase densities (:math:`\rho_p`) and phase viscosities (:math:`\mu_p`) are computed as a function
of pressure, temperature and the updated phase component fractions.

After calculating the phase fractions, phase component fractions, phase densities, phase viscosities,
and their derivatives with respect to pressure, temperature, and component fractions, the
:ref:`CompositionalMultiphaseFlow` then moves on to assembling the accumulation and flux terms.

Step 1: Computation of the phase fractions and phase component fractions (flash)
================================================================================
Stability test
-------------------------------

> The first step is to determine if provided mixture with total molar fractions :math:`z_c` is stable

Reviewer: ```suggestion
The first step is to determine if the provided mixture with total molar fractions :math:`z_c` is stable
```

as a single phase at the current pressure :math:`p` and temperature :math:`T`. However, this can only

> be confirmed with through stability testing.

Reviewer: ```suggestion
be confirmed through stability testing.
```

where :math:`h_i = \ln z_i + \ln \phi_i(z)` is a constant parameter dependent on the feed composition
:math:`z` and :math:`k` is an undetermined constant. This constant can be further incorporated into
the equation by defining the unnormalized trial phase moles :math:`Y_i` as

.. math::
  Y_i = \exp(-k)y_i

which reduces the stationarity criterion to

.. math::
  \ln Y_i + \ln \phi_i(y) - h_i = 0

with the mole fractions :math:`y_i` related to the trial phase moles :math:`Y_i` by

.. math::
  y_i = Y_i/\sum_jY_j

With the two starting mixtures, the stationarity condition is solved using successive substitution to
determine the stationary points. If both initial states converge to a solution which has :math:`g(y)\geq 0`
then the mixture is deemed to be stable, otherwise it is deemed unstable.

Phase labeling
----------------------------------------
Once it is confirmed that the fluid with composition :math:`z` is stable as a single phase at the current
pressure and temperature, it must be labeled as either 'liquid' or 'vapor'. This is necessary only to apply
the correct relative permeability function for calculating the phase's flow properties. The properties of the

> fluid (density, viscosity, enthalpy) are unchaged by the assignment of the label.

Author: ```suggestion
fluid (density, viscosity) are unchanged by the assignment of the label.
```
